### DISCUSSION TITLE
JupyterLite demo — Working With Github files

### DISCUSSION THREAD

**Discussion by psychemedia** [jupyterlite/jupyterlite#868] [Q&A]
The Github extension in the JupyterLite demo site provides a way of browsing and loading files from across a named Github user's repos:

<img width="871" alt="image" src="https://user-images.githubusercontent.com/82988/200597147-0580b382-fd89-4d56-a576-aca76e7a5498.png">

What is the envisaged use of this in the demo context?

I can:

- browse files;
- open and execute files

but I can't seem to:

- save edits in that file path to browser storage and be warned there is a diff with the repo view;
- save copies of the file into any other path in the file browser;
- copy over files from the browsed repo view into local browser storage.

As a read/exit/execute demo, it's quite nifty, but it can also be a frustrating if a user thinks they might be able to actually grab copies of files into a persistent Jupyterlite context and make persistent browser storage changes to them, unless they download and then upload the files.

Or am I missing a trick somewhere?

**Comment by jtpio** [comment]
This is because the GitHub browser acts as a different `drive` than the default file browser. And talk to GitHub directly without going through the default in-browser contents manager.

A workaround could be to open the read-only Github notebooks and save them with "Save As".
